File: app/src/mock/java/io/zenandroid/greenfield/dagger/FlickrServiceModule.java
package io.zenandroid.greenfield.dagger;

import dagger.Binds;
import dagger.Module;
import io.zenandroid.greenfield.service.FlickrService;
import io.zenandroid.greenfield.service.MockFlickrService;

/**
 * created by acristescu
 */
@Module
public abstract class FlickrServiceModule {

	@Binds
	abstract FlickrService provideBBCService(MockFlickrService bbcService);
}

File: app/src/mock/java/io/zenandroid/greenfield/dagger/FlickrServiceModule.kt
package io.zenandroid.greenfield.dagger

import dagger.Binds
import dagger.Module
import io.zenandroid.greenfield.service.FlickrService
import io.zenandroid.greenfield.service.MockFlickrService

/**
 * created by acristescu
 */
@Module
abstract class FlickrServiceModule {

    @Binds
    internal abstract fun provideFlickrService(bbcService: MockFlickrService): FlickrService
}
